### PR TITLE
[13.0] Create module sale_order_carrier_auto_assign - alpha

### PR DESCRIPTION
**Pull request by grindtildeath**
This module assigns automatically delivery carrier on sale order confirmation.

### CODE REVIEW

**Comment by grindtildeath** [comment]
@ivantodorovich @pedrobaeza @kongrattapong @nicomacr @cubells @chienandalu @carlosdauden As you guys were last contributors on module sale_order_type, can you please check ea817bf ? :pray: :hugs: 

Also please let me know if I should better extract this commit in a separated PR.

**Comment by pedrobaeza** [comment]
@grindtildeath I think this module should go to OCA/delivery-carrier, so with that, you don't have the problem. Anyway, I think the default was removed previously by a reason. Other thing you can do is to force the sale.order view to use in `Form`.

**Review by jbaudoux — APPROVED** [review]
LGTM

File: sale_order_carrier_auto_assign/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            "test_carrier_auto_assign"
+        ):
+            return super().action_confirm()
+        self._add_delivery_carrier_on_confirmation()
+        return super().action_confirm()
+
+    def _add_delivery_carrier_on_confirmation(self):
+        """Automatically add delivery.carrier on sale order confirmation"""
+        for order in self:
+            if order.carrier_id or any(line.is_delivery for line in order.order_line):
```
Comment (jbaudoux) [inline sale_order_carrier_auto_assign/models/sale_order.py:22]: I would use here "if order.delivery_set"
This is used to display on the SO form if the delivery method is set.

**Review by jbaudoux — CHANGES_REQUESTED** [review]

**Review by jgrandguillaume — APPROVED** [review]
LGTM thx

**Comment by jgrandguillaume** [comment]
I suggest to merge as it's alpha.

**Pull request merged**